egs/formosa/s5/run.sh

```diff
@@ -0,0 +1,249 @@
+   2500 20000 data/train data/lang exp/mono_ali exp/tri1 || exit 1;
+
+  # align tri1
+  steps/align_si.sh --cmd "$train_cmd" --nj $num_jobs \
+    data/train data/lang exp/tri1 exp/tri1_ali || exit 1;
+
+  # decode tri1
+  (
+  utils/mkgraph.sh data/lang_test exp/tri1 exp/tri1/graph || exit 1;
+  steps/decode.sh --cmd "$decode_cmd" --config conf/decode.config --nj $num_jobs \
+    exp/tri1/graph data/test exp/tri1/decode_test
+  )&
+
+fi
+
+# tri2
+if [ $stage -le 2 ]; then
+
+  echo "$0: train tri2 model"
+  # train tri2 [delta+delta-deltas]
+  steps/train_deltas.sh --cmd "$train_cmd" \
+   2500 20000 data/train data/lang exp/tri1_ali exp/tri2 || exit 1;
+
+  # align tri2b
+  steps/align_si.sh --cmd "$train_cmd" --nj $num_jobs \
+    data/train data/lang exp/tri2 exp/tri2_ali || exit 1;
+
+  # decode tri2
+  (
+  utils/mkgraph.sh data/lang_test exp/tri2 exp/tri2/graph
+  steps/decode.sh --cmd "$decode_cmd" --config conf/decode.config --nj $num_jobs \
+    exp/tri2/graph data/test exp/tri2/decode_test
+  )&
+
+fi
+
+# tri3a
+if [ $stage -le 3 ]; then
+
+  echo "$-: train tri3 model"
+  # Train tri3a, which is LDA+MLLT,
+  steps/train_lda_mllt.sh --cmd "$train_cmd" \
+   2500 20000 data/train data/lang exp/tri2_ali exp/tri3a || exit 1;
+
+  # decode tri3a
+  (
+  utils/mkgraph.sh data/lang_test exp/tri3a exp/tri3a/graph || exit 1;
+  steps/decode.sh --cmd "$decode_cmd" --nj $num_jobs --config conf/decode.config \
+    exp/tri3a/graph data/test exp/tri3a/decode_test
+  )&
+
+fi
+
+# tri4
+if [ $stage -le 4 ]; then
+
+  echo "$0: train tri4 model"
+  # From now, we start building a more serious system (with SAT), and we'll
+  # do the alignment with fMLLR.
+  steps/align_fmllr.sh --cmd "$train_cmd" --nj $num_jobs \
+    data/train data/lang exp/tri3a exp/tri3a_ali || exit 1;
+
+  steps/train_sat.sh --cmd "$train_cmd" \
+    2500 20000 data/train data/lang exp/tri3a_ali exp/tri4a || exit 1;
+
+  # align tri4a
+  steps/align_fmllr.sh  --cmd "$train_cmd" --nj $num_jobs \
+    data/train data/lang exp/tri4a exp/tri4a_ali
+
+  # decode tri4a
+  (
+  utils/mkgraph.sh data/lang_test exp/tri4a exp/tri4a/graph
+  steps/decode_fmllr.sh --cmd "$decode_cmd" --nj $num_jobs --config conf/decode.config \
+    exp/tri4a/graph data/test exp/tri4a/decode_test
+  )&
+
+fi
+
+# tri5
+if [ $stage -le 5 ]; then
+
+  echo "$0: train tri5 model"
+  # Building a larger SAT system.
+  steps/train_sat.sh --cmd "$train_cmd" \
+    3500 100000 data/train data/lang exp/tri4a_ali exp/tri5a || exit 1;
+
+  # align tri5a
+  steps/align_fmllr.sh --cmd "$train_cmd" --nj $num_jobs \
+    data/train data/lang exp/tri5a exp/tri5a_ali || exit 1;
+
+  # decode tri5
+  (
+  utils/mkgraph.sh data/lang_test exp/tri5a exp/tri5a/graph || exit 1;
+  steps/decode_fmllr.sh --cmd "$decode_cmd" --nj $num_jobs --config conf/decode.config \
+     exp/tri5a/graph data/test exp/tri5a/decode_test || exit 1;
+  )&
+
+fi
+
+if [ $stage -le 6 ]; then
+
+  local/run_cleanup_segmentation.sh
```

Comment:
it looks to me like maybe you copied the librispeech recipe or something like that.
Did you actually check that the cleanup was helpful?  It's kind of slow, and would be better to leave out if it doesn't help

Reply:
the average duration of utterances in training set is about 20s, so I copied the script to make them shorter. After processing, it can be reduced to 10s, but there is no significant improvement in CER. I will remove this step.
```
# %WER 16.83 [ 36305 / 215718, 4772 ins, 10810 del, 20723 sub ] exp/chain/tdnn_1a_sp/decode_test/cer_9_0.0
# %WER 16.44 [ 35459 / 215718, 4216 ins, 11278 del, 19965 sub ] exp/chain/tdnn_1b_sp/decode_test/cer_10_0.0
```